### PR TITLE
fix : timeout issue for argocd and vault for public clouds

### PR DESCRIPTION
**Pull request by jokestax**
To resolve the issue where the argocd-controller starts before the argo-cd server and encounters a missing secret key error, I've added a function to reload it:

### CODE REVIEW

**Review by fharper — COMMENTED** [review]
Seems to work for me for Argo CD, but not Vault

**Comment by fharper** [comment]
So my initial test was with k3d, which was wrong as it's not using the API but the runtime for now 🤦‍♂️

Will test again with a public cloud like Civo.

**Review by fharper — DISMISSED** [review]
I was able to create a Civo cluster on the first try.

File: Dockerfile [hunk 1]
```diff
@@ -1,37 +1,45 @@
...
-  apk add git openssh aws-cli
-
-RUN mkdir -p /root/.ssh
-RUN ssh-keyscan github.com >> /root/.ssh/known_hosts
-RUN ssh-keyscan gitlab.com >> /root/.ssh/known_hosts
 
 EXPOSE 8081
 
-# Command to run
+
```
Comment (jairoFernandez) [inline Dockerfile]: remove extra line

File: Dockerfile [hunk 2]
```diff
@@ -1,37 +1,45 @@
...
+RUN mkdir -p /root/.ssh \
+    && ssh-keyscan github.com >> /root/.ssh/known_hosts \
+    && ssh-keyscan gitlab.com >> /root/.ssh/known_hosts
+
+
+RUN curl -LO https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl \
+    && chmod +x ./kubectl \
+    && mv ./kubectl /usr/local/bin/kubectl
+
+
```
Comment (jairoFernandez) [inline Dockerfile]: remove extra line

File: Dockerfile [hunk 3]
```diff
@@ -1,37 +1,45 @@
...
+        git \
+        openssh \
+        aws-cli \
+        curl
+
+RUN mkdir -p /root/.ssh \
+    && ssh-keyscan github.com >> /root/.ssh/known_hosts \
+    && ssh-keyscan gitlab.com >> /root/.ssh/known_hosts
+
+
```
Comment (jairoFernandez) [inline Dockerfile]: eliminate the additional line, only one line between statements

File: internal/controller/argocd.go [hunk 4]
```diff
@@ -176,7 +178,7 @@ func (clctrl *ClusterController) DeployRegistryApplication() error {
 		if err != nil {
 			return err
 		}
-
+		// log.Info().Msg(kcfg.RestConfig)
```
Comment (jairoFernandez) [inline internal/controller/argocd.go]: 🧹

**Review by jairoFernandez — DISMISSED** [review]
just a few comments

**Review by fharper — APPROVED** [review]

**Pull request merged**